src/script_opt/ScriptOpt.cc

```diff
@@ -437,6 +454,21 @@ static void analyze_scripts_for_ZAM() {
 
     auto pfs = std::make_shared<ProfileFuncs>(funcs, nullptr, true);
 
+    if ( analysis_options.profile_ZAM ) {
+#ifdef ENABLE_ZAM_PROFILE
+        blocks = std::make_unique<BlockAnalyzer>(funcs);
+        const auto prof_filename = "zprof.log";
+        analysis_options.profile_file = fopen(prof_filename, "w");
+        if ( ! analysis_options.profile_file ) {
+            fprintf(stderr, "warning: cannot create ZAM profiling log %s\n", prof_filename);
+            analysis_options.profile_ZAM = false;
```

Comment:
If someone explicitly asked for profiling, seems hard-exit is less surprising, particularly with stderr redirection or so you might never know it didn't actually profile.